### PR TITLE
Rails Budget

### PR DESCRIPTION
**Pull request by killy10o10**
##  In this milestone: 
- created a rails app where users can create a category and add a transaction.
- the app has a splash screen
- users can see the total amount of spending they have on each category
- wrote specs for models and routes
- followed design guidelines by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) on Behance

> [video presentation](https://www.loom.com/share/c9b47e7752204cfda2ce45c932d68128)
rails-budget-spec

## Happy reviewing!!🥳🥂

### CODE REVIEW

File: app/views/categories/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                  <span class="fw-bold fs-4"><%= category.name %></span>
+                  <span class="fw-bold fs-4 total text-black"><%= category.expenses.sum(:amount) %></span>
+                </div>
+                <span class="time fst-italic"><%= category.created_at.strftime('%d %B %Y at %I:%M %p') %></span>
+              </div>
+            </div> <br>
+      <%end%>
+    <%end%>
+  <% end %>
+  <button class="btn btn-success"><%= link_to "NEW CATEGORY", new_category_path %></button>
```
Comment (codepantha) [inline app/views/categories/index.html.erb]: - [x] Please let's try to stick to the Behance design's guideline by placing the NEW CATEGORY button at the bottom of the page and making sure that it takes up the entire width as shown below 👇 
- [x] You have used bootstrap `btn-success` which doesn't match the specified background color for the buttons. Kindly use the specified background color `#5fb523` for the button
## Design Guideline
image

## Your Implementation
image

File: app/views/categories/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        <span class="fw-bold fs-4">$<%= @total_expenses %></span>
+        <div class="d-flex gap-2 align-items-baseline justify-content-center mb-2">
+          <%= link_to "EDIT", edit_category_path(@category), class: "cat-command", id: "edit" %>
+          <%= button_to "DELETE", category_path(@category), method: :delete, class: "cat-command", id: "delete" %> 
+        </div>
+      </div>
+      <span class="time fst-italic"><%= @category.created_at.strftime('%d %B %Y at %I:%M %p') %></span>
+    </div>
+  </div> <br>
+  <button class="btn btn-success"><%= link_to "NEW TRANSACTION", new_expense_path %></button>
```
Comment (codepantha) [inline app/views/categories/show.html.erb]: - [x] Just like I stated earlier, let's try to stick to the Behance design's guideline by placing the NEW TRANSACTION button at the bottom of the page and making sure that it takes up the entire width as shown below 👇
- [x] You have used bootstrap `btn-success` which doesn't match the specified background color for the buttons. Kindly use the specified background color `#5fb523` for the button


image

File: app/views/categories/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      <div class="category-name border-bottom d-flex justify-content-between align-items-center">
+        <span class="fw-bold fs-4"><%= @category.name %></span>
+        <span class="fw-bold fs-4">$<%= @total_expenses %></span>
+        <div class="d-flex gap-2 align-items-baseline justify-content-center mb-2">
+          <%= link_to "EDIT", edit_category_path(@category), class: "cat-command", id: "edit" %>
+          <%= button_to "DELETE", category_path(@category), method: :delete, class: "cat-command", id: "delete" %> 
+        </div>
+      </div>
+      <span class="time fst-italic"><%= @category.created_at.strftime('%d %B %Y at %I:%M %p') %></span>
+    </div>
```
Comment (codepantha) [inline app/views/categories/show.html.erb]: - [x] When I click on a specific category that has transactions, I expect to see the transactions in that particular category listed on the page. 
In the attached image, you can see that I have done a few transactions but they are not listed. I think you missed this particular feature. Kindly implement it 🙏 
image
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @killy10o10 , 

You have done an amazing job so far 👏!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ Rspec tests 📌 
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

- [x] One of the requirements for the demo presentation is to highlight a piece of the code that you find interesting or are proud of making. Kindly re-record the video and add this vital part 👍 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Deepakdanger — APPROVED** [review]
Hi @killy10o10, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit:


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**